### PR TITLE
Add tests for using PEP560 with classes implemented in C.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Based on tests from #4878.

### CODE REVIEW

File: Modules/_testcapimodule.c [hunk 1]
```diff
@@ -5053,6 +5053,96 @@ recurse_infinitely_error_init(PyObject *self, PyObject *args, PyObject *kwds)
...
+generic_init(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    PyGenericObject *o = (PyGenericObject*) self;
+
+    PyObject *item;
+    if (!PyArg_UnpackTuple(args, "Generic", 1, 1, &item)) {
+        return -1;
+    }
+
+    o->item = item;
```
Comment (gvanrossum) [inline Modules/_testcapimodule.c]: Is there a DECREF missing here? [PyArg_UnpackTuple returns a borrowed item](https://docs.python.org/3/c-api/arg.html#c.PyArg_UnpackTuple). Also there might be a value already here so a DECREF would also be needed (before the assignment). Unless I misunderstand something (totally possible, I feel very rusty in this area :-).

**Review by gvanrossum — COMMENTED** [review]
Otherwise LG.

File: Modules/_testcapimodule.c [hunk 2]
```diff
@@ -5053,6 +5053,96 @@ recurse_infinitely_error_init(PyObject *self, PyObject *args, PyObject *kwds)
 }
 
 
+typedef struct {
+    PyObject_HEAD
+    PyObject *item;
+} PyGenericObject;
+
+
+/* Test PEP 560 slots */
```
Comment (1st1) [inline Modules/_testcapimodule.c]: Outdated comment

File: Modules/_testcapimodule.c (hunk 1 above)
Comment (1st1) [reply Modules/_testcapimodule.c]: I think `Py_INCREF(item)` is needed.

Comment (1st1) [reply Modules/_testcapimodule.c]: But for the purposes of this test we should be able to just remove this function and `tp_init` slot.

**Review by 1st1 — APPROVED** [review]

**Review by ilevkivskyi — APPROVED** [review]
Thank you for adding these tests! I just have one suggestion to make the test closer to how these methods will be used in typical code. The idea is that there will be typically a pair of classes:
```python
class Generic:
    def __class_getitem__(cls, item):
        return GenericAlias(item)

class GenericAlias:
    def __init__(self, item):
        self.item = item
    def __mro_entries__(self, bases):
        return (item,)
```
but this is just a minor suggestion, you can merge it as is if you prefer.

**Comment by ilevkivskyi** [comment]
Thanks!

(If we agree to change ``__class_getitem__`` to classmethod, then I will update this test later.)

**Pull request merged**